plugins/out_stackdriver/stackdriver.c

```diff
@@ -2498,14 +2502,26 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
         flb_upstream_conn_release(u_conn);
         FLB_OUTPUT_RETURN(FLB_RETRY);
     }
-    payload_size = flb_sds_len(payload_buf);
+
+    out_payload_buf = (flb_sds_t) payload_buf;
+    out_payload_size = flb_sds_len(payload_buf);
+    if (ctx->compress != NULL && strcmp(ctx->compress, "gzip") == 0) {
+        ret = flb_gzip_compress((void *) payload_buf, flb_sds_len(payload_buf),
```

Comment:
[flb_gzip_compress](https://github.com/fluent/fluent-bit/blob/ba49c5baa998f5187c6224ebcaf400de9f98370a/src/flb_gzip.c#L85) returns a buffer allocated with `flb_malloc` which is not compatible with `flb_sds_destroy`

Reply:
Thanks! Done.